aten/src/ATen/native/native_functions.yaml

```diff
@@ -8557,6 +8557,28 @@
   dispatch:
     CPU, CUDA: linalg_slogdet_out
 
+- func: linalg_eig(Tensor self) -> (Tensor eigenvalues, Tensor eigenvectors)
+  python_module: linalg
+  variants: function
+  dispatch:
+    CPU, CUDA: linalg_eig
```

Comment:
Functional variant here and below redispatch, so CompositeExplicitAutograd would also have worked too. But even better, consider making this structured kernel https://github.com/pytorch/rfcs/blob/rfc-0005/RFC-0005-structured-kernel-definitions.md

Reply:
Structured kernel support seems like a good follow-up PR.